packages/service-library/src/servicelib/archiving_utils.py

```diff
@@ -15,9 +17,29 @@ def _strip_undecodable_in_path(path: Path) -> Path:
     return Path(str(path).encode(errors="replace").decode("utf-8"))
 
 
-def _full_file_path_from_dir_and_subdirs(dir_path: Path) -> Iterator[Path]:
+def _get_exclude_regex_rules(exclude_paths: List[Path]) -> List[str]:
+    results: Deque[str] = deque()
+    for path in exclude_paths:
+        if path.is_absolute():
+            results.append(rf"^{path}.*")  # match starts with
+        else:
+            results.append(rf"(.*){path}$")  # match ends with
+    return list(results)
+
+
+def _should_exclude(regex_rules: List[str], path: Path) -> bool:
+    for rule in regex_rules:
+        if re.match(rule, f"{path}"):
+            return False
+    return True
+
+
+def _full_file_path_from_dir_and_subdirs(
+    dir_path: Path, exclude_paths: List[Path] = None
+) -> Iterator[Path]:
+    regex_rules = _get_exclude_regex_rules(exclude_paths if exclude_paths else [])
```

Comment:
I am not sure I understand what these changes are supposed to be doing.
maybe you could create some well named constants with the regex inside that describe what they do? Why not calling just is_absolute instead of doing that stuff with the regexes?